components/scream/src/share/field/field.hpp

```diff
@@ -178,11 +178,11 @@ class Field {
 
   // Set the field to a constant value (on host or device)
   template<typename T, HostOrDevice HD = Device>
-  void deep_copy (const T value);
+  void deep_copy (const T value) const;
```

Comment:
I am a little worried that we're abusing the meaning of `const` here. Strictly speaking, a `const` View can have its data mutated because Views use reference semantics, but this is never going to be obvious to anyone who looks at this thing to try to gain an understanding of what changes what.

I don't think this warrants a change. Just complaining. :-)

Reply:
Yeah, it would also remove the need for `mutable` for the MPI request vectors.